### DISCUSSION TITLE
Using python package "lbg" instead of dpcloudserver will make the program easier to maintain

### DISCUSSION THREAD

**Discussion by xiaoyeqiannian** [deepmodeling/dpdispatcher#305] [General]
"lbgcore" has been well encapsulated in the python library "lbg". We can replace the work of dpcloudserver in dpdispatcher by "import lbgcore". In the future, the new featcher and bug fixes of "lbg" will be updated to dpdispatcher in a more timely manner, what’s your opinion? @njzjz

**Comment by njzjz** [comment]
It looks good to me.

**Comment by njzjz** [comment]
Do you think it's a better idea to implement a plugin of `dpdispatcher` in the `lbg` package? For the possible implementation of the plugin, there is an example in [dpdata](https://github.com/deepmodeling/dpdata#plugins).

**Reply by xiaoyeqiannian** [reply]
Good idea! If I implement a class similar to Format in dpdata on dpdispatcher to replace the Machine class, It will need to test in slurm, pbs and etc... Is there a more efficient method for testing?

**Reply by njzjz** [reply]
There have already been some [tests](https://github.com/deepmodeling/dpdispatcher/tree/master/ci) for slurm and PBS.

Indeed, I don't think `Machine` needs to be changed. It has a plugin-like `__init_subclass__` method.
https://github.com/deepmodeling/dpdispatcher/blob/f30244c2e0fee6fe849732bd2b8a32727414fcba/dpdispatcher/machine.py#L111-L117

However, we need to import these classes. Thus we just need this
```py
ep_name = "dpdispatcher.plugins"  # or just dpdispatcher is good
try:
    eps = metadata.entry_points(group=ep_name)
except TypeError:
    eps = metadata.entry_points().get(ep_name, [])
for ep in eps:
    plugin = ep.load()
```

**Reply by xiaoyeqiannian** [reply]
oh, i see
